native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ResponseGenerator.java

```diff
@@ -97,10 +96,9 @@ static void getDataFromService(ExecutionContext executionContext, BObject servic
         } else {
             executionContext.setTypeName(service.getType().getName());
             for (int i = 0; i < selections.size(); i++) {
-                BMap<BString, Object> selection = (BMap<BString, Object>) selections.get(i);
-                boolean isFragment = selection.getBooleanValue(IS_FRAGMENT_FIELD);
-                BObject subNode = selection.getObjectValue(NODE_FIELD);
-                if (isFragment) {
+                BObject subNode = (BObject) selections.get(i);
+                BString typeName = StringUtils.fromString(subNode.getType().getName());
+                if (typeName.equals(FRAGMENT_NODE)) {
```

Comment:
Usually, when comparing two objects using the `equals` method, use the constant first. It will reduce the possibility of throwing an NPE.
```suggestion
                if (FRAGMENT_NODE.equals(typeName)) {
```